test/metadata/main.go

```diff
@@ -0,0 +1,306 @@
+		os.Exit(1)
+	}
+
+	// fmt.Println(string(createFieldData))
+
+	var fieldOperation *common.XMLData
+
+	if err := json.Unmarshal(createFieldData, &fieldOperation); err != nil {
+		slog.Error("Error marshalling testOperation.json", "error", err)
+		os.Exit(1)
+	}
+
+	operation := &common.XMLData{
+		XMLName:     "createMetadata",
+		Children:    []common.XMLSchema{objectOperation, fieldOperation},
+		SelfClosing: false,
+	}
+
+	res2, err := sfc.CreateMetadata(context.Background(), operation, tok)
+	if err != nil {
+		slog.Debug("err", "err", err)
+	}
+
+	fmt.Println("Field Operation Result", res2)
+}
+
+func getCreateObjectOperationDefinition() *common.XMLData {
+	fieldType := &common.XMLData{
+		XMLName:     "type",
+		Children:    []common.XMLSchema{common.XMLString("Text")},
+		SelfClosing: false,
+	}
+	nameFieldLabel := &common.XMLData{
+		XMLName:     "label",
+		Children:    []common.XMLSchema{common.XMLString("Test Object Name")},
+		SelfClosing: false,
+	}
+
+	nameField := &common.XMLData{
+		XMLName:     "nameField",
+		Children:    []common.XMLSchema{fieldType, nameFieldLabel},
+		SelfClosing: false,
+	}
+
+	deploymentStatus := &common.XMLData{
+		XMLName:     "deploymentStatus",
+		Children:    []common.XMLSchema{common.XMLString("Deployed")},
+		SelfClosing: false,
+	}
+
+	sharingModel := &common.XMLData{
+		XMLName:     "sharingModel",
+		Children:    []common.XMLSchema{common.XMLString("ReadWrite")},
+		SelfClosing: false,
+	}
+
+	fullName := &common.XMLData{
+		XMLName:     "fullName",
+		Children:    []common.XMLSchema{common.XMLString("TestObject13__c")},
+		SelfClosing: false,
+	}
+
+	ObjecLabel := &common.XMLData{
+		XMLName:     "label",
+		Children:    []common.XMLSchema{common.XMLString("Test Object 13")},
+		SelfClosing: false,
+	}
+
+	pluralLabel := &common.XMLData{
+		XMLName:     "pluralLabel",
+		Children:    []common.XMLSchema{common.XMLString("Test Objects 13")},
+		SelfClosing: false,
+	}
+
+	metadata := &common.XMLData{
+		XMLName:     "metadata",
+		Attributes:  []*common.XMLAttributes{{Key: "xsi:type", Value: "CustomObject"}},
+		Children:    []common.XMLSchema{fullName, ObjecLabel, pluralLabel, nameField, deploymentStatus, sharingModel},
+		SelfClosing: false,
+	}
+
+	operation := &common.XMLData{
+		XMLName:     "createMetadata",
+		Children:    []common.XMLSchema{metadata},
+		SelfClosing: false,
+	}
+
+	// fmt.Println(operation.ToXML())
+
+	// jsonData, err := json.MarshalIndent(operation, "", "  ")
+	// if err != nil {
+	// 	slog.Error("Error marshalling operation", "error", err)
+	// }
+
+	// fmt.Println(string(jsonData))
+
+	return operation
+}
+
+func getCreateFieldOperation() *common.XMLData {
+	metadata := &common.XMLData{
+		XMLName:    "metadata",
+		Attributes: []*common.XMLAttributes{{Key: "xsi:type", Value: "CustomField"}},
+		Children: []common.XMLSchema{
+			&common.XMLData{
+				XMLName: "fullName",
+				Children: []common.XMLSchema{
+					common.XMLString("TestObject13__c.Comments__c"),
```

Comment:
I think having the `XMLString` type makes constructing XMLData more tedious than necessary. Is there a strong benefit you see in using it vs just a string type here?

Reply:
Children is array of XMLSchema which is an interface that requires ToXML() for parsing purpose of children. 
Children can be string or another XMLData, and when parsing through, parse operates by `child.ToXML()` and appends to an array.

Otherwise I need to make `Children` to be interface{}, the in side the parser, check switch case to check the child component which I thought was uneccessary complexicty.